File: benchmarks/pandas/bench_dataframe_reflected_arith.py
"""Benchmark: dataframe_reflected_arith — DataFrame.radd / rsub / rmul / rdiv."""
import json, time
import numpy as np
import pandas as pd

ROWS = 10_000
WARMUP = 5
ITERATIONS = 30

df = pd.DataFrame({
    "a": np.arange(ROWS) * 1.5,
    "b": (np.arange(ROWS) % 100) + 1.0,
    "c": np.arange(ROWS) * 0.25,
})

for _ in range(WARMUP):
    df.radd(10)
    df.rsub(1000)
    df.rmul(3)
    df.rdiv(100)

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.radd(10)
    df.rsub(1000)
    df.rmul(3)
    df.rdiv(100)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "dataframe_reflected_arith",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_dataframe_reflected_arith.ts
/**
 * Benchmark: dataframe_reflected_arith — dataFrameRadd / dataFrameRsub / dataFrameRmul / dataFrameRdiv.
 * Outputs JSON: {"function": "dataframe_reflected_arith", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { DataFrame, dataFrameRadd, dataFrameRsub, dataFrameRmul, dataFrameRdiv } from "../../src/index.ts";

const SIZE = 50_000;
const WARMUP = 5;
const ITERATIONS = 30;

const df = DataFrame.fromColumns({
  a: Array.from({ length: SIZE }, (_, i) => i * 1.5),
  b: Array.from({ length: SIZE }, (_, i) => (i % 100) + 1),
  c: Array.from({ length: SIZE }, (_, i) => i * 0.25),
});

for (let i = 0; i < WARMUP; i++) {
  dataFrameRadd(df, 10);
  dataFrameRsub(df, 1000);
  dataFrameRmul(df, 3);
  dataFrameRdiv(df, 100);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  dataFrameRadd(df, 10);
  dataFrameRsub(df, 1000);
  dataFrameRmul(df, 3);
  dataFrameRdiv(df, 100);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dataframe_reflected_arith",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
